app/modules/api/project.py

```diff
@@ -30,4 +31,31 @@ def project_post(json_content):
         return jsonify(
             {"id": project.id, "team_id": project.team_id, "name": project.name}
         )
-    return api_error_helpers.could_not_create("project")
\ No newline at end of file
+    return api_error_helpers.could_not_create("project")
```

Comment:
1. I'd move this to Team and just dump EEEEVERYTHING, i looked for a charting library that could support exactly this, also, it doesn't make lots of sense to query _every_ project in the team file to display the complete chart. Right now (mvp) we only care about the whole project, not project by project.
2. The @req_helper.api_check_json() decorator will error out if no json was provided, i think? however, the params are optional... so you'd need to send a valid empty json?
3. Week.fromstring() only accepts isoweek format (ie “2011W08”), would be nice to accept other alternatives like: a relative week count 0 being the current week | a full iso date | the ordinal week #
4. Also on the not before, what about defaulting to the current week, i think we only care about future weeks (tho lemme recheck that)
5. what about adding a not_after? and defaulting to.. idk, six months from now.
6. I would really like the relative week scheme since it would play veery very nicely with sliders. That way i can request week 0 to week 0 (inclusive), and get just the current week, or week 0 to week 3 to get the next weeks n stuff. plz fix

Reply:
Also on the dumping everything, I already made the aggregation pipeline to summarize and average several weeks on the front end when a range longer than 1 week is provided. I suggest we use that and let the backend dump all... _OR_ por that to python and save some bandwith at the expense of processing time?

Reply:
PS. looks like i places this on the wrong line, this is pertaining to `def project_schedules(json_content, project_id: int)`

Reply:
So responses...

1. On the summary pipeline, can you show me where the code is? I'll add both a call to dump everything and one for summarization separately (that way we could even cache).
2. That's... Accurate. I've been sending "{}", though I could add a parameter to ignore. That or, if we switch to GET requests with URL params issue goes away.
3.  I'm not too sure about the ordinal (we want to hide that from the user) but using an ISO date would make sense. I'll switch to that. I thought the ISO format would make the most sense for consistency, but we don't have a library for that frontend so I see what you're saying
4. LMK
5. Fair enough. Wasn't sure if it would be useful
6. Huh... In that case, using only "ordinals" for 3 would make a lot of sense. Let me do that then

Reply:
Just found an issue with relative dates. Server will probably be on a different timezone than users (hopefully UTC). I'll stick with absolute dates, but use ISO dates instead of weeks for easier interface